### ISSUE TITLE
Why does the official log of TPN-TSM have no validation results for dataset STHV1? And a big difference between the verification results just after training and the results of testing after loading from the current checkpoint

### ISSUE THREAD

**Issue by ParkHUQ** [open-mmlab/mmaction2#831]
Here is the content of the run script that we reproduced:
./tools/dist_train.sh configs/recognition/tpn/tpn_tsm_r50_1x1x8_150e_sthv1_rgb.py 8 --work-dir work_dirs/tpn_tsm_r50_1x1x8_150e_sthv1_rgb --validate --seed 0 --deterministic

./tools/dist_test.sh configs/recognition/tpn/tpn_tsm_r50_1x1x8_150e_sthv1_rgb.py work_dirs/tpn_tsm_r50_1x1x8_150e_sthv1_rgb.py/epoch_45.pth  8 --eval top_k_accuracy mean_class_accuracy --average-clips prob

Taking the 45th epoch as an example,  results as follows:
The first one is the verification results after training and the other is the test results by loading from the 45th epoch checkpoint.
![图片](https://user-images.githubusercontent.com/58662352/115966928-63857200-a562-11eb-81d9-a2afa68a3693.png)
![图片](https://user-images.githubusercontent.com/58662352/115977099-41faa980-a5a7-11eb-9c37-a50d5d05ca24.png)

Changes in the configuration file:
1. Turn 'FrameSelector' to 'RawFrameDecode' in train/val/test pipeline;
2. Add 'filename_tmpl='{:05d}.jpg' in data's dict.

Could you please tell us the reason and how to deal with it? Thanks so much.

****

**Comment by congee524** [comment]
Sometimes we didn't do validation when training the model to save the training time.

validation and testing have different views. We sample 8 clips and do CenterCrop in cross-validation, while we sample 16 clips and do ThreeCrop in testing.

**Comment by ParkHUQ** [comment]
Thanks for answering. Do you mean the official top-1 ACC 50.8 is obtained by testing?

Our test settings are the same as validation settings, just like this:
    test=dict(
        type=dataset_type,
        ann_file=ann_file_test,
        data_prefix=data_root_val,
        filename_tmpl='{:05d}.jpg',
        **pipeline=val_pipeline**))
We just used the test model to get the results by validation settings because the results of validation when training were not normal. Anything do you know about this situation?

**Comment by congee524** [comment]
Yes, the most papers will report the result with multi-views.

If you use val_pipeline, then the testing result should be the same as the validation result.

**Comment by ParkHUQ** [comment]
But the pictures in the first comment suggests a big difference in results by the two ways. What the problem might be?

**Comment by dreamerlin** [comment]
You can try #833 this testing with twice sample and num_clips=8
Details can be found in https://github.com/open-mmlab/mmaction2/pull/241

**Comment by congee524** [comment]
The low precision in cross-validation may be caused by `ColorJitter` in val_pipeline which should not be applied when testing.

Meanwhile, you can follow the config that just is modified. We replace `num_clips=16` with `num_clips=8 && twice_sample`.
Usually, this gets better results. You can refer to #241 for their difference.

**Comment by ParkHUQ** [comment]
Thx for your advice, but it seems like the problem not solved. 

The validation results of TPN-TSM were not normal when training, but sometimes the results came to be normal after resuming and rerun the model. Those validation results are similar to the results by loading from the current checkpoint when testing.

**Comment by dreamerlin** [comment]
It is wired ... I suggest u clone the current master branch and run with our provided dist script again.

**Comment by ParkHUQ** [comment]
To be honest, we have repeated the clone several times,  but still have such a problem. It seems like other networks have no obvious such problems, just only TPN...
